### DISCUSSION TITLE
Github Actions: Spring Boot Postgres Gradle

### DISCUSSION THREAD

**Discussion by zfz7** [community/community#25193] [Actions]
<p><strong>Goal</strong> : Run ./gradlew test after each push to master</p>
<p><strong>Setup</strong> : Fairly simple Spring boot setup with Postgres Database</p>
<p><strong>Issue:</strong> Spring can’t connect to DB</p>
<p>See <a href="https://github.com/zfz7/spring-jwt-microservice/tree/83c78bba30bf5be4f9e3d2dbcedfa37312d59c2c" rel="nofollow noopener">this</a> commit to see the last effort I made to get github actions working.</p>
<p>The following code is primarly based on <a href="https://github.com/actions/example-services/blob/master/.github/workflows/postgres-service.yml" rel="nofollow noopener">this</a> github action.</p>
<p>Test.yml</p>
<pre><code>name: Gradle Build

on:
  push:
    branches:
      - master
  pull_request:

jobs:
  vm-jobs:
    runs-on: ubuntu-latest

    services:
      postgres:
        image: postgres:12.2
        env:
          POSTGRES_USER: user
          POSTGRES_PASSWORD: pass
          POSTGRES_DB: spring-jwt
        ports:
          # will assign a random free host port
          - 5432/tcp
        # needed because the postgres container does not provide a healthcheck
        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

    steps:
      - uses: actions/checkout@v1

      - name: Set up JDK 1.8
        uses: actions/setup-java@v1
        with:
          java-version: 1.8

      - name: Run tests
        run: ./gradlew test
</code></pre>
<p>application.properties</p>
<pre><code>spring.datasource.url=jdbc:postgresql://localhost:5432/spring-jwt
spring.jpa.generate-ddl=true
spring.datasource.username=user
spring.datasource.password=pass
</code></pre>
<p>Errors I am seeing:</p>
<pre><code>AuthApplicationTests &gt; Successfull login FAILED
    java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java:132
        Caused by: org.springframework.beans.factory.BeanCreationException at AbstractAutowireCapableBeanFactory.java:1796
            Caused by: org.hibernate.service.spi.ServiceException at AbstractServiceRegistryImpl.java:275
                Caused by: org.hibernate.HibernateException at DialectFactoryImpl.java:100
</code></pre>
<p>I am 99% sure these errors arre because Spring Boot can find the database.</p>
<p>Any pointers would be great!</p>
<p>-Daniel</p>

**Comment by zfz7** [comment, marked as answer]
<p>I fixed it. See <a href="https://github.com/zfz7/spring-jwt-microservice/tree/0cbb37b7d7131f8a2f3653d86a83602cd015c568" rel="nofollow noopener">this</a> commit for the first passing test. To be honest I don’t really know what fixed it (it was a combination of things). If you somehow stumble upon this I would recommend first learning how to use docker and docker compose. There are way more tutorials on that topic.</p>

**Reply by wwarreniv** [reply]
4 years later, I came across this issue and it seems that removing the "" from your port mapping fixed it; thats what worked for me anyway. Thanks for posting the commit number!

**Comment by BrightRan** [comment]
<p><a class="mention" href="/zfz7">@zfz7</a> ,</p>
<p>Glad to hear that the problem has been solved.</p>
<p>If you have any other question about this ticket, feel free to tell us. Thanks.</p>
